WordPress/gutenberg
# Add ability to preview template in post editor for non administrators

## What?

Since we found an [alternative solution to fix the issue that editors were not able to select a template in the post editor](https://github.com/WordPress/gutenberg/pull/58485) anymore, this PR now is focused on bringing the ability to preview a post to non administrator roles also. 

Closing: #58037

## Why?


 - Updating the `wp_template` post type rest API permissions to allow anyone with the `edit_posts` role to view it. 
 - Updaing the `template part` block to have a new preview only mode for anyone that doesn't have permission to edit the template


## Testing Instructions


1. Make sure your theme supports at least 1 additional custom template besides the default single post template.
2. Open the post editor with a user account that is an Editor or lower role
3. See that you are again able to switch between the available templates

### Testing Instructions for Keyboard


## Screenshots or screencast
